### PR TITLE
Refactor tests_influences:

### LINKED ISSUE #388: Testing torch related modules is non-deterministic
Tests for the influence module, in particular those testing the lissa and arnoldi inversion method randomly fail with AssertionErrors,
due to the difference not being in the desired tolerance. Due to the stochastic nature of lissa and training routines needed for arnoldi tests, we should make those test deterministic.

### PR DESCRIPTION
**Pull request by schroedk**
### Description

This PR closes #388

### Changes

* use dataclasses for test case definition
* make test deterministic

### Checklist

- [x] Wrote Unit tests (if necessary)
- ~~[ ] Updated Documentation (if necessary)~~
- ~~[ ] Updated Changelog~~
- ~~[ ] If notebooks were added/changed, added boilerplate cells are tagged with `"nbsphinx":"hidden"`~~

### CODE REVIEW

File: tests/influence/test_influences.py [hunk 1]
```diff
@@ -77,6 +78,13 @@ def analytical_linear_influences(
     return result
 
 
+@pytest.fixture(scope="function")
+def deterministic_seed():
```
Comment (AnesBenmerzoug) [inline tests/influence/test_influences.py:82]: We already have [2 fixtures](https://github.com/appliedAI-Initiative/pyDVL/blob/develop/tests/conftest.py#L61-L76) that are used to seed random numbe generators.
Perhaps they can be combined (at least `pytorch_seed` and `deterministic_seed`?

File: tests/influence/test_influences.py [hunk 2]
```diff
@@ -413,3 +476,15 @@ def test_influences_arnoldi(
     )
 
     assert np.allclose(direct_influence, low_rank_influence, rtol=1e-1)
+
+
+@pytest.mark.torch
+@pytest.mark.parametrize("run", [1, 2])
+def test_deterministic_seed(deterministic_seed, run):
```
Comment (AnesBenmerzoug) [inline tests/influence/test_influences.py:483]: is this test really needed? It tests PyTorch's behaviour which is something that we do not control anyway.

**Comment by AnesBenmerzoug** [comment]
In case you want to test an algorithm/method that is fast enough to run and without pinning the seed of the random number generators, we have a special [`tolerate`](https://github.com/appliedAI-Initiative/pyDVL/blob/develop/tests/conftest.py#L374) fixture for that than can be used to run a particular call multiple times and allow a maximum number of failures.

File: tests/influence/test_influences.py (hunk 2 above)
Comment (schroedk) [reply tests/influence/test_influences.py:483]: Yeah, agree, will remove it

File: tests/influence/test_influences.py (hunk 1 above)
Comment (schroedk) [reply tests/influence/test_influences.py:82]: We could add torch.use_deterministic_algorithms to pytorch_seed. This should work then. What do you think?

**Review by AnesBenmerzoug — APPROVED** [review]
